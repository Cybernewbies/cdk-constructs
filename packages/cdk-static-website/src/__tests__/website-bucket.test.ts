import { Stack, RemovalPolicy } from '@aws-cdk/core';
import 'jest-cdk-snapshot';

import { WebsiteBucket } from '../website-bucket';

test('default setup', (): void => {
  const stack = new Stack();

  new WebsiteBucket(stack, 'WebsiteBucket', {
    bucketName: 'testbucket',
    disableUpload: true,
  });

  expect(stack).toMatchCdkSnapshot();
});

test('removalPolicy = DESTROY', (): void => {
  const stack = new Stack();

  new WebsiteBucket(stack, 'WebsiteBucket', {
    bucketName: 'testbucket',
    disableUpload: true,
    removalPolicy: RemovalPolicy.DESTROY,
  });

  expect(stack).toMatchCdkSnapshot({
    ignoreAssets: true,
  });
});
