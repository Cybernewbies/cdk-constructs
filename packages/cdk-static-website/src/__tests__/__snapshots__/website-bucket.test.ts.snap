// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default setup 1`] = `
Object {
  "Resources": Object {
    "WebsiteBucket3C288C20": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "BucketName": "testbucket",
        "WebsiteConfiguration": Object {
          "ErrorDocument": "error.html",
          "IndexDocument": "index.html",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
    "WebsiteBucketOriginAccessIdentity717BE380": Object {
      "Properties": Object {
        "CloudFrontOriginAccessIdentityConfig": Object {
          "Comment": Object {
            "Fn::Join": Array [
              "",
              Array [
                "CloudFront OriginAccessIdentity for ",
                Object {
                  "Ref": "WebsiteBucket3C288C20",
                },
              ],
            ],
          },
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
    "WebsiteBucketPolicy65FBF6CB": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "WebsiteBucket3C288C20",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "CanonicalUser": Object {
                  "Fn::GetAtt": Array [
                    "WebsiteBucketOriginAccessIdentity717BE380",
                    "S3CanonicalUserId",
                  ],
                },
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "WebsiteBucket3C288C20",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "WebsiteBucket3C288C20",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
}
`;

exports[`removalPolicy = DESTROY 1`] = `
Object {
  "Parameters": Any<Object>,
  "Resources": Object {
    "SingletonLambdaCloudcomponentsEmptyBucketCustomResource09B44FFE": Object {
      "DependsOn": Array [
        "SingletonLambdaCloudcomponentsEmptyBucketCustomResourceServiceRoleDefaultPolicyFA08A909",
        "SingletonLambdaCloudcomponentsEmptyBucketCustomResourceServiceRoleEE758E51",
      ],
      "Properties": Object {
        "Code": Any<Object>,
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "SingletonLambdaCloudcomponentsEmptyBucketCustomResourceServiceRoleEE758E51",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "SingletonLambdaCloudcomponentsEmptyBucketCustomResourceServiceRoleDefaultPolicyFA08A909": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
                "s3:DeleteObject*",
                "s3:PutObject*",
                "s3:Abort*",
              ],
              "Effect": "Allow",
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "WebsiteBucket3C288C20",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "WebsiteBucket3C288C20",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "SingletonLambdaCloudcomponentsEmptyBucketCustomResourceServiceRoleDefaultPolicyFA08A909",
        "Roles": Array [
          Object {
            "Ref": "SingletonLambdaCloudcomponentsEmptyBucketCustomResourceServiceRoleEE758E51",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "SingletonLambdaCloudcomponentsEmptyBucketCustomResourceServiceRoleEE758E51": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "WebsiteBucket3C288C20": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "BucketName": "testbucket",
        "WebsiteConfiguration": Object {
          "ErrorDocument": "error.html",
          "IndexDocument": "index.html",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
    "WebsiteBucketEmptyBucketCustomResourceD9F7451B": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "BucketName": Object {
          "Ref": "WebsiteBucket3C288C20",
        },
        "ServiceToken": Object {
          "Fn::GetAtt": Array [
            "WebsiteBucketEmptyBucketEmptyBucketProviderframeworkonEventC2C38DD2",
            "Arn",
          ],
        },
      },
      "Type": "Custom::EmptyBucket",
      "UpdateReplacePolicy": "Delete",
    },
    "WebsiteBucketEmptyBucketEmptyBucketProviderframeworkonEventC2C38DD2": Object {
      "DependsOn": Array [
        "WebsiteBucketEmptyBucketEmptyBucketProviderframeworkonEventServiceRoleDefaultPolicy818946AD",
        "WebsiteBucketEmptyBucketEmptyBucketProviderframeworkonEventServiceRole14A08AC3",
      ],
      "Properties": Object {
        "Code": Any<Object>,
        "Description": "AWS CDK resource provider framework - onEvent (Default/WebsiteBucket/EmptyBucket/EmptyBucketProvider)",
        "Environment": Object {
          "Variables": Object {
            "USER_ON_EVENT_FUNCTION_ARN": Object {
              "Fn::GetAtt": Array [
                "SingletonLambdaCloudcomponentsEmptyBucketCustomResource09B44FFE",
                "Arn",
              ],
            },
          },
        },
        "Handler": "framework.onEvent",
        "Role": Object {
          "Fn::GetAtt": Array [
            "WebsiteBucketEmptyBucketEmptyBucketProviderframeworkonEventServiceRole14A08AC3",
            "Arn",
          ],
        },
        "Runtime": "nodejs10.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "WebsiteBucketEmptyBucketEmptyBucketProviderframeworkonEventServiceRole14A08AC3": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "WebsiteBucketEmptyBucketEmptyBucketProviderframeworkonEventServiceRoleDefaultPolicy818946AD": Object {
      "Properties": Object {
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "lambda:InvokeFunction",
              "Effect": "Allow",
              "Resource": Object {
                "Fn::GetAtt": Array [
                  "SingletonLambdaCloudcomponentsEmptyBucketCustomResource09B44FFE",
                  "Arn",
                ],
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "PolicyName": "WebsiteBucketEmptyBucketEmptyBucketProviderframeworkonEventServiceRoleDefaultPolicy818946AD",
        "Roles": Array [
          Object {
            "Ref": "WebsiteBucketEmptyBucketEmptyBucketProviderframeworkonEventServiceRole14A08AC3",
          },
        ],
      },
      "Type": "AWS::IAM::Policy",
    },
    "WebsiteBucketOriginAccessIdentity717BE380": Object {
      "Properties": Object {
        "CloudFrontOriginAccessIdentityConfig": Object {
          "Comment": Object {
            "Fn::Join": Array [
              "",
              Array [
                "CloudFront OriginAccessIdentity for ",
                Object {
                  "Ref": "WebsiteBucket3C288C20",
                },
              ],
            ],
          },
        },
      },
      "Type": "AWS::CloudFront::CloudFrontOriginAccessIdentity",
    },
    "WebsiteBucketPolicy65FBF6CB": Object {
      "Properties": Object {
        "Bucket": Object {
          "Ref": "WebsiteBucket3C288C20",
        },
        "PolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": Array [
                "s3:GetObject*",
                "s3:GetBucket*",
                "s3:List*",
              ],
              "Effect": "Allow",
              "Principal": Object {
                "CanonicalUser": Object {
                  "Fn::GetAtt": Array [
                    "WebsiteBucketOriginAccessIdentity717BE380",
                    "S3CanonicalUserId",
                  ],
                },
              },
              "Resource": Array [
                Object {
                  "Fn::GetAtt": Array [
                    "WebsiteBucket3C288C20",
                    "Arn",
                  ],
                },
                Object {
                  "Fn::Join": Array [
                    "",
                    Array [
                      Object {
                        "Fn::GetAtt": Array [
                          "WebsiteBucket3C288C20",
                          "Arn",
                        ],
                      },
                      "/*",
                    ],
                  ],
                },
              ],
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::S3::BucketPolicy",
    },
  },
}
`;
